torchgen/dest/ufunc.py

```diff
@@ -286,13 +286,14 @@ def compute_ufunc_cuda(g: NativeFunctionsGroup) -> str:
 
     # Next, build the conditionals
     sig = StructuredImplSignature(g, ufunc.kernel_name(g, DispatchKey.CUDA))
+    supported_dtypes = sorted(ufunctor_sigs.keys(), key=lambda x: str(x))
     dtype_cases = []
-    for dtype, inner_ufunctor_sigs in ufunctor_sigs.items():
+    for dtype in supported_dtypes:
```

Comment:
ugh, I'm confused, I thought in sufficiently recent versions of Python 3 dictionary traversal is deterministic. Is this actually not true?

Reply:
You're correct, dict traversal happens in the order of insertion. However, the order in which the dict is constructed here depends on iterating over this set:
https://github.com/pytorch/pytorch/blob/8c688f297165b3b56db993e0b40a0c14f84b9347/torchgen/model.py#L1048

Reply:
So, while we have landed this diff, perhaps we should replace the set with something like Dict[ScalarType, None]...